### PR TITLE
Clean name/dimensionless

### PR DESCRIPTION
**Pull request by DaluS**
# Update ! 

I'm baaaaaack, sorry for the delay. Here is a list of changes : 

## Fields file

* I reorganized fields in a way that seems more practical
* Verification of units 
* dimensionless units are now explicitly taken into account through units = '' ( None mean that it is not defined )
* There is a temporary section on the top for fields that are not definitive and introduced in models, as I struggle with adding them inside the model file (the system try to put them in the resolution order of the equations)

## Models 

* There is still a small number of models ( GK-reduced and GK ) for economy because I'm still struggling with the solvers
* I added non-economic models ( damped oscillators and lorenz system) for general testing 

## Multiple cockpit files  

* `TEST-CONVERGENCE` test the different solvers both quantitatively (on the damped oscillator) and qualitatively (on Lorenz)
* `BASIN-ATTRACTION-GK` use the potential of parallel system solving to look at the relation between initial and final value. It calculates the boundary between good and bad attractor in a GK model given (lambda,omega,d) when t=0

mygif

_the GIF : the color of each pixel is private debt value in time, the position of each pixel is the initial lambda/omega value. When pixel stay red, the trajectory is going to bad equilibrium. The shape of this frontier is found for each  initial value.  This has been calculated on runs with the homemade rk4, some runs are not working hence the white square. This gif thus contains 20*20*11 simulations with 20/0.005 timesteps, with one minute total computation). Ultimately the shape of the basin of attraction can be calculated using each layer of initial debt value, showing this type of lines.._

image

## What are the next things ? 

* Understanding why the RK8 Solver sometimes fail in case it should not 
* Understanding why the homemade RK4 is not working the intended way
* Include elements developed in the different cockpits files into pygemmes files
* Write correct presets so that unit test are on preset values
* Add more models of course 
* Review of the plots section to allow same units variables to be plot altogether
* Allowing multisectorial writing (AKA Matt's development)

### CODE REVIEW

**Comment by DaluS** [comment]
It's working well on my computer, I have absolutely no idea why these tests fail on github, can @Didou09 you check on your side ?

**Comment by DaluS** [comment]
The error comes from test multiple which was tried on a model name written explicitely, that had been changed previously. Next pr #116 solves it

File: pygemmes/tests/test_01_Hub.py [hunk 1]
```diff
@@ -29,6 +28,7 @@
 
 # library-specific
 sys.path.insert(0, _PATH_PCK)   # ensure Main comes from .. => add PYTHONPATH
+import pygemmes as pgm
```
Comment (Didou09) [inline pygemmes/tests/test_01_Hub.py:31]: This was already done in devel, are you sure you started from devel ?

File: pygemmes/_models/_model_GK-Reduced.py [hunk 2]
```diff
@@ -34,9 +34,8 @@
             'com': 'reduced 2variables dynamical expression'
         },
         'omega': {
-            'func': lambda itself=0, phillips=0, i=0, gamma=0: itself * phillips - (0.5)*i,
+            'func': lambda itself=0, phillips=0, i=0, gammai=0, alpha=0: itself * (phillips - (1 - gammai)*i-alpha),
```
Comment (Didou09) [inline pygemmes/_models/_model_GK-Reduced.py:37]: if gammai does not match a variable name the code won't be able to reckognize it in the dependence-searching

File: pygemmes/tests/test_01_Hub.py [hunk 3]
```diff
@@ -28,7 +29,6 @@
 
 # library-specific
 sys.path.insert(0, _PATH_PCK)   # ensure Main comes from .. => add PYTHONPATH
-import pygemmes as pgm
```
Comment (Didou09) [inline pygemmes/tests/test_01_Hub.py:31]: why remove it ? This part was to make sure the local version was loaded

**Review by Didou09 — COMMENTED** [review]
Commit 1:
---------------

* By convention, since I only have limited time, I only check what is inside folder `pygemmes/` (+ `setup.py` and `requirements.txt` for the packaging), so I won't check changes to `Cockpit.py` and all other files / folders not in `pygemmes/`. 
This way @DaluS you can do what best suits your needs outside of `pygemmes/` without me interfering, and I focus on the cor part of the package.

* I agree with the name changing of `_model_G_Reduced.py` to `_model_G-Reduced.py`, this is in accordance with the conventions discussed in issue #107 

* `_models/_def_fields.py`:  ok, i understand you're setting default units to '' instead of None and introducing 2 new parameters

* `_models/_model_GK.py` : ok, I understand you're introducing more complete functions for `phillips` and `I` and introducing new function `solvability`

I am ok with this first commit

**Review by Didou09 — COMMENTED** [review]
Commit 2:
--------------

* Overlooking `Cockpit.py`

* `_models/_def_fields.py`: ok, introducing new parameter `gamma`

* `_models/_model_GK-Reduced.py`: ok, introducing a brand new model

* `_models/_model_GK.py`: ok, making several functions more complete

I am ok with this commit

**Review by Didou09 — COMMENTED** [review]
Commit 3:
--------------

* Overlooking `Cockpit.py`

* `_core.Hub. FillCycles()`: ok, removing useless comment block + adding debugging print

* `_def_fields.py`: ok, replacing `zphi` by `zsolv` (to avoid naming conflicts ?), and adding `solvability`

* `_model_GK-Reduced-SolvProfit.py` : ok, introducing brand new model

* `_model_GK-Reduced.py`: ok, replacing `zkappa` by `zsolv`, adding solvability

* _model_GK.py : ok, replacing `zkappa` by `zsolv`


I am ok with this commit

**Review by Didou09 — COMMENTED** [review]
Commit 4:   #b8ac5acd77195cba0c6717b46bbdd439a3a71146
--------------

* Overlooking `Cockpit.py`

I am ok with this commit

**Comment by DaluS** [comment]
> * By convention, since I only have limited time, I only check what is inside folder `pygemmes/` (+ `setup.py` and `requirements.txt` for the packaging), so I won't check changes to `Cockpit.py` and all other files / folders not in `pygemmes/`.
>   This way @DaluS you can do what best suits your needs outside of `pygemmes/` without me interfering, and I focus on the cor part of the package.

Yes ! I think that's the best way ! 
There are "optional" libraries (to draw variable graph/basin of attraction), that we should add (the necessary libraries are the same as before)

**Review by Didou09 — COMMENTED** [review]
Commit 5: #0859e0fefd59a5476f8850fcd3120cb38e054c50
--------------

* Overlooking  `Cockpit.py`

* `_def_fields.py`: ok
    * passing `time` into `Numerical` group
    * Renaming `CORE` group to `Household`
    * Removing `beta`, `delta`, `omega`
    * Reorganising several groups of parameters
    * New `Consumption` group
    * Introducing new parameters `gammai`, `GDP`, `Recipes`

* `_model-MonoGEM.py`: not ok, brand new model BUT, the file name is not conform to the accepted conventions ! => will be corrected in future commits?

* `_model_GK.py`: ok, changing a bit equation on `W`

* `_model-MonoGEM.py`: ok, brand new model


I am ok with this commit, provided the wrong file is removed in future commits

**Review by Didou09 — COMMENTED** [review]
Commit 6 :   #e88f27dcdaf88c8170e15c2ab0ad0520cb56e723
----------------

* Overlooking `Cockpit.py` and `Cockpit2.py`

* `_def_fields.py`: ok, minor changes in the parameter library

* `_model_CES-Reduced.py`: ok, brand new model

* `_model_DampOscillator`.py: ok, brand new model

* `_model_GK-Reduced.py`: ok, minor changes to existing model

* `_model_LorenzSystem.py`: ok, brand new model

* `_model_MonoGEM.py`: ok, removed model

* `_utilities/_class_checks.py`: ok, minor indentation change

I am ok with this commit

**Review by Didou09 — COMMENTED** [review]
Commit 7:     #1e99d6d84702d010b3e88d5740cd144eb04c965d
---------------

* Overlooking `Cockpit2.py`

I am ok with this commit

**Review by Didou09 — COMMENTED** [review]
Commit 8: #a950efca3019278562ee39e78df5f0343d234c6e
--------------

* Overlooking `BASIN-ATTRACTION-GK.py`,  `Cockpit.py`, `NotIncluded/Cockpit2.py`, `NotIncluded/Cockpit3GK.py`, `TEST-CONVERGENCE.py`

* Several models moved to `NotIncluded/`: ok

* `NotIncluded/test_02_Hub_Multiple.py`: not ok a priori, this means we are removing a previously working unit test, which is bad practice but can be justified if temporary. Will depend on future commits

* Several models modified: ok (`DampOscillator`, `GK-Reduced`, `GK`, `LorenzSystem`)

* Non-regression reference test files removed: not ok a priori, depends whether they are properly replaced by updated ones in later commits

* `pygemmes/tests/test_01_Hub.py`: not ok
    * import pygemmes moved to top: not ok because it cannot ensure that the local version is loaded (will have to be corrected in future commits)
    * Removed replacement of underscore by dash in model name: not ok because it forces conventions, will have to be corrected in future commits

I am ok with this commit, with the following caveats that have to be corrected in future commits:
* `test_02_Hub_Multiple.py` is necessary
* Non-regression tests are necessary
* `test_01_Hub.py` will have to be corrected backwards

**Review by Didou09 — COMMENTED** [review]
Commit 9 (last):   #3c0b2e717a780af44d63ef099f080e0e519c7c64
---------------------

* Overlooking `Cockpit.py`

* `NotIncluded/_model_GK.py`: ok, but may be necessary for unit tests ?

* `_model_GK-Reduced.py`: ok, minor change

* `test_01_Hub.py`: ok, import pygemmes bacj in right place (remove one caveat from previous commit)

I am ok with this commit, but there are still 2 remaining caveats: `test_02_Hub_Multiple.py` and non-regression tests.

+ Unit tests are not passing, so with your approval @DaluS I will fix unit tests and solve the 2 remaining caveats

**Comment by DaluS** [comment]
> All susequent commits were validated in PR #124

I close this PR in consequence

**Pull request closed without merging**